plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1842,6 +2042,7 @@ public Optional<ConnectorOutputMetadata> finishInsert(
                     .get(context -> commitInsertOperation(session, handle, sourceTableHandles, isolationLevel, dataFileInfos, readVersion, context.getAttemptCount()));
             writeCommitted = true;
             writeCheckpointIfNeeded(session, handle.tableName(), handle.location(), handle.readVersion(), handle.metadataEntry().getCheckpointInterval(), commitVersion);
+            cacheTableMetadataAsync(handle.tableName().getSchemaName(), handle.tableName().getTableName(), Optional.ofNullable(handle.metadataEntry().getDescription()), commitVersion, handle.metadataEntry().getSchemaString());
 
             if (isCollectExtendedStatisticsColumnStatisticsOnWrite(session) && !computedStatistics.isEmpty() && !dataFileInfos.isEmpty()) {
```

Comment:
If `CREATE OR REPLACE` lands likely first, this is a reminder to add handling for it as well.